test/clj/rems/api/test_categories.clj

```diff
@@ -0,0 +1,179 @@
+        (is (int? id-1))
+
+        (let [result-2 (api-call :post "/api/categories"
+                                 (merge create-category-data
+                                        {:category/children [{:category/id id-1}]})
+                                 +test-api-key+ owner)
+              id-2 (:id result-2)]
+          (is (true? (:success result-2)))
+          (is (int? id-2))
+
+          (let [category (api-call :get (str "/api/categories/" id-2) nil
+                                   +test-api-key+ owner)
+                expected (merge create-category-data
+                                join-organization
+                                {:id id-2
+                                 :category/children [{:category/id id-1
+                                                      :category/title (get create-category-data :category/title)}]})]
+            (is (= category expected))))))
+
+    (testing "enrich unknown categories"
+      (let [owner "owner"
+            result (api-call :post "/api/categories"
+                             (merge create-category-data
+                                    {:category/children [{:category/id 9999999}]})
+                             +test-api-key+ owner)
+            id (:id result)]
+        (is (true? (:success result)))
+        (is (int? id))
+
+        (let [category (api-call :get (str "/api/categories/" id) nil
+                                 +test-api-key+ owner)
+              expected (merge create-category-data
+                              join-organization
+                              {:id id
+                               :category/children [{:category/id 9999999
+                                                    :category/title {:fi "Tuntematon kategoria"
+                                                                     :sv "Okänd kategori"
+                                                                     :en "Unknown category"}}]})]
+          (is (= category expected)))))))
+
+(deftest categories-api-update-test
+  (let [owner "owner"
+        org-owner "organization-owner1"
+        create-category-data {:organization {:organization/id "organization1"}
+                              :category/title {:fi "integraatiotesti"
+                                               :sv "integrationstest"
+                                               :en "integration test"}
+                              :category/description {:fi "integraatiotesti"
+                                                     :sv "integrationstest"
+                                                     :en "integration test"}
+                              :category/children []}
+        update-category-data {:category/title {:fi "integraatiotesti 2"
+                                               :sv "integrationstest 2"
+                                               :en "integration test 2"}}
+        join-organization {:organization {:organization/id "organization1"
```

Comment:
Should we add one more test that checks like the security tests we have elsewhere, that only an org owner can modify/delete a category of their own org?